neuracore/ml/utils/policy_inference.py

```diff
@@ -217,9 +217,9 @@ def _process_image_data(
         Returns:
             MaskableData containing batched image tensors and attention masks.
         """
-        channels = 1 if is_depth else 3
-        values = np.zeros((1, max_len, channels, 224, 224))
-        mask = np.zeros((1, max_len))
+        # Use lists to collect images - no need to know size upfront
+        image_list = []
+        mask_list = []
```

Comment:
Why remove pre-construction? 
More efficient to pre-constrict the arrays than constantly append to lists